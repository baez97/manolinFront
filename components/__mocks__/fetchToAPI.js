const fetchToAPI = jest.fn();
export default fetchToAPI;