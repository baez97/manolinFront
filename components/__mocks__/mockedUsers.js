const documents = [
    {
        "_id": {
            "$oid": "5c7ab0c331f6240fbc248f6e"
        },
        "name": "Ana",
        "password": "$2a$08$Er1/dLfJSdvJojynt6Ykj.UvK..Hd.9P1Cs7Akkyb0QdX92pEXePm",
        "email": null,
        "turnWithDates": [
            {
                "day": 1,
                "month": 1,
                "year": 2019,
                "weekday": 2,
                "turn": "M"
            },
            {
                "day": 2,
                "month": 1,
                "year": 2019,
                "weekday": 3,
                "turn": "M"
            },
            {
                "day": 3,
                "month": 1,
                "year": 2019,
                "weekday": 4,
                "turn": "T"
            },
            {
                "day": 4,
                "month": 1,
                "year": 2019,
                "weekday": 5,
                "turn": "T"
            }
        ]
    },
    {
        "_id": {
            "$oid": "5c7c0e8a5434ee0287d3e6b6"
        },
        "name": "Mar√≠a",
        "password": "$2a$08$W2mbg6LEEIxx81h9p9VFj.RF6UASLiou/bLQul7y3ITgFhwbdNmWC",
        "email": null,
        "turnWithDates": [
            {
                "day": 1,
                "month": 1,
                "year": 2019,
                "weekday": 2,
                "turn": "M"
            },
            {
                "day": 2,
                "month": 1,
                "year": 2019,
                "weekday": 3,
                "turn": "M"
            },
            {
                "day": 3,
                "month": 1,
                "year": 2019,
                "weekday": 4,
                "turn": "T"
            }
        ]
    }
];

module.exports = documents;