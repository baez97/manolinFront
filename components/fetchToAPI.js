import { BACKEND_IP } from '../config';

function fetchToAPI(urlString, options) {
    return fetch( BACKEND_IP + urlString, options);
}

export default fetchToAPI;