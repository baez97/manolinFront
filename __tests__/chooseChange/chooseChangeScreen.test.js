import React from 'react';
import renderer from 'react-test-renderer';
import ChooseChangeScreen from '../../screens/chooseChangeScreen'
import fetchToAPI from '../../components/fetchToAPI';
import mockedUsers from '../../components/__mocks__/mockedUsers';

jest.mock('../../components/dateUtils');
jest.mock('../../components/fetchToAPI');
jest.mock('../../components/homeComponents/changes/changeView');
jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo'); 

describe("ChooseChangeScreen", () => {
    describe("Constructor", () => {
        it("Gets the Navigation Params and Binds the methods", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn()
                }
            };

            new ChooseChangeScreen(mockedProps);

            expect( mockedProps.navigation.getParam ).toBeCalled();
        });
    });

    describe("ComponentDidMount", () => {
        it("Catches a fetch error", async() => {
            fetchToAPI.mockReturnValueOnce( new Promise((res, rej) => {
                rej("ERR");
            }));

            const mockedProps = {
                navigation: {
                    getParam: jest.fn(() => ({ owner_id: "id" }))
                },
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);
            CCScreen.setState = jest.fn();
            
            await CCScreen.componentDidMount();

            expect( CCScreen.setState ).toBeCalledWith({ error: true })
        });

        it("Calls to setState if everything was fine", async () => {
            fetchToAPI.mockReturnValueOnce( new Promise((res, rej) => {
                res({ json: () => ("MOCKED_USER")})
            }));

            const mockedProps = {
                navigation: {
                    getParam: jest.fn(() => ({ owner_id: "id" }))
                },
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);
            CCScreen.setState = jest.fn();
            
            await CCScreen.componentDidMount();

            expect( CCScreen.ownerNurse ).toBe( "MOCKED_USER" );
        });
    });

    describe("Functions", () => {
        it("WantsToChange works fine", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn()
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);

            // MOCKING DEPENDENCIES OF THE FUNCTION
            CCScreen.showModal = jest.fn();
            CCScreen.changerNurse = { name : "MOCKED_NAME" };
            CCScreen.ownerNurse   = { name : "MOCKED_NAME" };
            CCScreen.change       = { day  : "MOCKED_DAY",
                                      turn : "MOCKED_TURN" };
            const mockedTurn = {
                day: "MOCKED_DAY",
                turn: "MOCKED_TURN",
            }

            // EXECUTING THE TESTED FUNCTION
            CCScreen.wantsToChange(mockedTurn);

            // ASSERTING
            expect( CCScreen.showModal ).toBeCalled();
        });

        it("ConfirmChange works fine", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn()
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);

            // MOCKING DEPENDENCIES OF THE FUNCTION
            CCScreen.showConfirmation = jest.fn();
            CCScreen.change           = "MOCKED_CHANGE"
            CCScreen.socket           = { emit: jest.fn() };
            CCScreen.changerNurse     = { name : "MOCKED_NAME" };
            const mockedTurn = {
                day: "MOCKED_DAY",
                turn: "MOCKED_TURN",
            }

            // EXECUTING THE TESTED FUNCTION
            CCScreen.confirmChange(mockedTurn);

            // ASSERTING
            expect( CCScreen.showConfirmation ).toBeCalled();
        });

        it("ShowModal works fine", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn()
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);

            // MOCKING DEPENDENCIES OF THE FUNCTION
            CCScreen.setState = jest.fn();

            // EXECUTING THE TESTED FUNCTION
            CCScreen.showModal("message", "turn");

            // ASSERTING
            expect( CCScreen.setState ).toBeCalled();
        });

        it("HideModal works fine", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn()
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);

            // MOCKING DEPENDENCIES OF THE FUNCTION
            CCScreen.setState = jest.fn();

            // EXECUTING THE TESTED FUNCTION
            CCScreen.hideModal();

            // ASSERTING
            expect( CCScreen.setState ).toBeCalled();
        });

        it("ShowConfirmation works fine", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn()
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);

            // MOCKING DEPENDENCIES OF THE FUNCTION
            CCScreen.setState = jest.fn();
            CCScreen.ownerNurse = { name: "MOCKED_NAME" }

            // EXECUTING THE TESTED FUNCTION
            CCScreen.showConfirmation();

            // ASSERTING
            expect( CCScreen.setState ).toBeCalled();
        });

        it("ShowConfirmation works fine", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn()
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);

            // MOCKING DEPENDENCIES OF THE FUNCTION
            CCScreen.setState = jest.fn();
            CCScreen.ownerNurse = { name: "MOCKED_NAME" }

            // EXECUTING THE TESTED FUNCTION
            CCScreen.showConfirmation();

            // ASSERTING
            expect( CCScreen.setState ).toBeCalled();
        });

        it("HideConfirmation works fine", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn(() => "MOCKED_TOKEN"),
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);

            // MOCKING DEPENDENCIES OF THE FUNCTION
            const mockedFn = jest.fn();
            CCScreen.props.navigation.navigate = mockedFn;

            // EXECUTING THE TESTED FUNCTION
            CCScreen.hideConfirmation();

            // ASSERTING
            expect( mockedFn ).toBeCalled();
        });

        it("SetMonth works fine", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn(() => "MOCKED_TOKEN"),
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);

            // MOCKING DEPENDENCIES OF THE FUNCTION
            CCScreen.setState = jest.fn();

            // EXECUTING THE TESTED FUNCTION
            CCScreen.setMonth("MOCKED_MONTH");

            // ASSERTING
            expect( CCScreen.setState ).toBeCalledWith(
                { currentMonthIndex: "MOCKED_MONTH" }
            );
        });

        it("DoNothing does nothing", () => {
            const mockedProps = {
                navigation: {
                    getParam: jest.fn(() => "MOCKED_TOKEN"),
                }
            };

            const CCScreen = new ChooseChangeScreen(mockedProps);
            CCScreen.setState = jest.fn();

            CCScreen.doNothing();

            expect( CCScreen.setState ).not.toBeCalled();
        });
    });

    describe("View", () => {
        it("Shows 'Cargando...' when it is loading", async () => {
            fetchToAPI.mockReturnValue(
                new Promise((res, rej) => {})
            );
            const navigation = {
                getParam: jest.fn( key => "TOKEN" )
            }

            const rendered = await renderer.create(
                <ChooseChangeScreen navigation={navigation}/>
            );

            const loadingView = rendered.toJSON().children[0];
            const loadingText = loadingView.children[0];
            expect( loadingText ).toBe("Cargando...");
            expect( rendered    ).toMatchSnapshot();
        });

        it("Shows 'Se ha producido un error...' when the fetch failed", async () => {
            
            fetchToAPI.mockReturnValue(
                new Promise((res, rej) => rej("ERROR"))
            );

            const mockedToken = "VALUE";

            const navigation = {
                getParam: jest.fn( key => mockedToken )
            }

            const rendered = await renderer.create(
                <ChooseChangeScreen navigation={navigation}/>
            );

            await rendered.root._fiber.stateNode.componentDidMount()

            const loadingView = rendered.toJSON().children[0];
            const loadingText = loadingView.children[0];
            expect( loadingText ).toBe("Se ha producido un error");
            expect( rendered    ).toMatchSnapshot();
        });

        it("Matches the Snapshot when everything is okay", async () => {
            
            fetchToAPI.mockReturnValue(
                new Promise((res, rej) => 
                    res({ json: () => mockedUsers[0]})
                )
            );

            const mockedChange = {
                owner: "Ana",
                owner_id: "5c7ab0c331f6240fbc248f6e",
                day: 5,
                month: 2,
                year: 2019,
                weekday: 1,
                turn: "M",
                type: "change"
            };

            const navigation = {
                getParam: jest.fn( key => {
                    if ( key === "changerNurse" )
                        return mockedUsers[1];
                    else if ( key === "change")
                        return mockedChange;
                    else
                        return "MOCKED_VALUE"
                })
            }

            const rendered = await renderer.create(
                <ChooseChangeScreen navigation={navigation}/>
            );

            await rendered.root._fiber.stateNode.componentDidMount()

            expect( rendered    ).toMatchSnapshot();
        });

        it("Shows 'Turno' when currentMonthIndex has not solved yet", async () => {
            
            fetchToAPI.mockReturnValue(
                new Promise((res, rej) => 
                    res({ json: () => mockedUsers[0]})
                )
            );

            const mockedChange = {
                owner: "Ana",
                owner_id: "5c7ab0c331f6240fbc248f6e",
                day: 5,
                month: 2,
                year: 2019,
                weekday: 1,
                turn: "M",
                type: "change"
            };

            const navigation = {
                getParam: jest.fn( key => {
                    if ( key === "changerNurse" )
                        return mockedUsers[1];
                    else if ( key === "change")
                        return mockedChange;
                    else
                        return "MOCKED_VALUE"
                })
            }


            const CCScreen = new ChooseChangeScreen({navigation: navigation, changerNurse: mockedUsers[0]});
            CCScreen.ownerNurse = mockedUsers[1];
            CCScreen.state = {
                userLoaded: true,
                error: false,
                currentMonthIndex: 14,
                nurses: mockedUsers
            }

            const view = CCScreen.render();
            const monthView = view.props.children[2];
            const monthText = monthView.props.children;
            expect ( monthText ).toBe("Turno");
        });
    });
});