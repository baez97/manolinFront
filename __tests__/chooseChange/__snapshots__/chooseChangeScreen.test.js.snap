// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseChangeScreen View Matches the Snapshot when everything is okay 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bee6ef",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 70,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#332554",
        "fontFamily": "montserrat-extra-bold",
        "fontSize": 40.7291678090925,
        "marginTop": 87.5,
        "textAlign": "center",
        "width": 712.5,
      }
    }
  >
    ¿Qué turno le propones a Ana que haga a cambio?
  </Text>
  <Text
    style={
      Object {
        "color": "#332554",
        "fontFamily": "montserrat-extra-bold",
        "fontSize": 38.95833442608516,
        "marginTop": 87.5,
        "textAlign": "center",
        "width": 712.5,
      }
    }
  >
    Febrero
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 17.5,
        "width": 712.5,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 35,
          "marginTop": 78.75,
        }
      }
    />
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": 35.5,
          "elevation": 15,
          "overflow": "hidden",
          "width": 531.5,
        }
      }
    />
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackDropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onBackDropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 35.5,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "paddingBottom": 43.75,
              "paddingLeft": 26.25,
              "paddingRight": 26.25,
              "paddingTop": 26.25,
              "width": 602,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2990f7",
                "fontFamily": "montserrat-extra-bold",
                "fontSize": 40.7291678090925,
                "marginBottom": 17.5,
                "textAlign": "center",
              }
            }
          >
            
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <ExponentLinearGradient
              colors={
                Array [
                  4278439238,
                  4278429498,
                ]
              }
              endPoint={
                Array [
                  0.65,
                  1,
                ]
              }
              startPoint={
                Array [
                  0.55,
                  0,
                ]
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 35.5,
                  "elevation": 15,
                  "height": 141.5,
                  "justifyContent": "center",
                  "marginTop": 61.25,
                  "width": 442.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "montserrat-extra-bold",
                    "fontSize": 38.95833442608516,
                    "textAlign": "center",
                  }
                }
              >
                Confirmar
              </Text>
            </ExponentLinearGradient>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <ExponentLinearGradient
              colors={
                Array [
                  4279286482,
                  4278481116,
                ]
              }
              endPoint={
                Array [
                  0.65,
                  1,
                ]
              }
              startPoint={
                Array [
                  0.55,
                  0,
                ]
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 35.5,
                  "elevation": 15,
                  "height": 141.5,
                  "justifyContent": "center",
                  "marginTop": 61.25,
                  "width": 442.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "montserrat-extra-bold",
                    "fontSize": 38.95833442608516,
                    "textAlign": "center",
                  }
                }
              >
                Cancelar
              </Text>
            </ExponentLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackDropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onBackDropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 35.5,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "paddingBottom": 43.75,
              "paddingLeft": 26.25,
              "paddingRight": 26.25,
              "paddingTop": 26.25,
              "width": 602,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#03a73a",
                "fontFamily": "montserrat-extra-bold",
                "fontSize": 40.7291678090925,
                "marginBottom": 17.5,
                "textAlign": "center",
              }
            }
          >
            
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 350,
                "width": 350,
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <ExponentLinearGradient
              colors={
                Array [
                  4278439238,
                  4278429498,
                ]
              }
              endPoint={
                Array [
                  0.65,
                  1,
                ]
              }
              startPoint={
                Array [
                  0.55,
                  0,
                ]
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 35.5,
                  "elevation": 15,
                  "height": 141.5,
                  "justifyContent": "center",
                  "marginTop": 61.25,
                  "width": 442.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "montserrat-extra-bold",
                    "fontSize": 38.95833442608516,
                    "textAlign": "center",
                  }
                }
              >
                Okey
              </Text>
            </ExponentLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`ChooseChangeScreen View Shows 'Cargando...' when it is loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bee6ef",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 70,
    }
  }
>
  <Text
    style={
      Object {
        "marginTop": 50,
      }
    }
  >
    Cargando...
  </Text>
</View>
`;

exports[`ChooseChangeScreen View Shows 'Se ha producido un error...' when the fetch failed 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bee6ef",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 70,
    }
  }
>
  <Text
    style={
      Object {
        "marginTop": 50,
      }
    }
  >
    Se ha producido un error
  </Text>
</View>
`;

exports[`ChooseChangeScreen View Shows 'Turno' when currentMonthIndex has not solved yet 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bee6ef",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 70,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#332554",
        "fontFamily": "montserrat-extra-bold",
        "fontSize": 40.7291678090925,
        "marginTop": 87.5,
        "textAlign": "center",
        "width": 712.5,
      }
    }
  >
    ¿Qué turno le propones a Ana que haga a cambio?
  </Text>
  <Text
    style={
      Object {
        "color": "#332554",
        "fontFamily": "montserrat-extra-bold",
        "fontSize": 38.95833442608516,
        "marginTop": 87.5,
        "textAlign": "center",
        "width": 712.5,
      }
    }
  >
    Febrero
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 17.5,
        "width": 712.5,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 35,
          "marginTop": 78.75,
        }
      }
    />
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": 35.5,
          "elevation": 15,
          "overflow": "hidden",
          "width": 531.5,
        }
      }
    />
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackDropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onBackDropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 35.5,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "paddingBottom": 43.75,
              "paddingLeft": 26.25,
              "paddingRight": 26.25,
              "paddingTop": 26.25,
              "width": 602,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2990f7",
                "fontFamily": "montserrat-extra-bold",
                "fontSize": 40.7291678090925,
                "marginBottom": 17.5,
                "textAlign": "center",
              }
            }
          >
            
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <ExponentLinearGradient
              colors={
                Array [
                  4278439238,
                  4278429498,
                ]
              }
              endPoint={
                Array [
                  0.65,
                  1,
                ]
              }
              startPoint={
                Array [
                  0.55,
                  0,
                ]
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 35.5,
                  "elevation": 15,
                  "height": 141.5,
                  "justifyContent": "center",
                  "marginTop": 61.25,
                  "width": 442.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "montserrat-extra-bold",
                    "fontSize": 38.95833442608516,
                    "textAlign": "center",
                  }
                }
              >
                Confirmar
              </Text>
            </ExponentLinearGradient>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <ExponentLinearGradient
              colors={
                Array [
                  4279286482,
                  4278481116,
                ]
              }
              endPoint={
                Array [
                  0.65,
                  1,
                ]
              }
              startPoint={
                Array [
                  0.55,
                  0,
                ]
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 35.5,
                  "elevation": 15,
                  "height": 141.5,
                  "justifyContent": "center",
                  "marginTop": 61.25,
                  "width": 442.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "montserrat-extra-bold",
                    "fontSize": 38.95833442608516,
                    "textAlign": "center",
                  }
                }
              >
                Cancelar
              </Text>
            </ExponentLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackDropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onBackDropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 35.5,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "paddingBottom": 43.75,
              "paddingLeft": 26.25,
              "paddingRight": 26.25,
              "paddingTop": 26.25,
              "width": 602,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#03a73a",
                "fontFamily": "montserrat-extra-bold",
                "fontSize": 40.7291678090925,
                "marginBottom": 17.5,
                "textAlign": "center",
              }
            }
          >
            
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 350,
                "width": 350,
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <ExponentLinearGradient
              colors={
                Array [
                  4278439238,
                  4278429498,
                ]
              }
              endPoint={
                Array [
                  0.65,
                  1,
                ]
              }
              startPoint={
                Array [
                  0.55,
                  0,
                ]
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 35.5,
                  "elevation": 15,
                  "height": 141.5,
                  "justifyContent": "center",
                  "marginTop": 61.25,
                  "width": 442.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "montserrat-extra-bold",
                    "fontSize": 38.95833442608516,
                    "textAlign": "center",
                  }
                }
              >
                Okey
              </Text>
            </ExponentLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
